src/main/java/com/programmers/springweekly/domain/voucher/VoucherType.java

```diff
@@ -0,0 +1,32 @@
+package com.programmers.springweekly.domain.voucher;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public enum VoucherType {
+    FIXED("fixed"),
+    PERCENT("percent");
+
+    private final String type;
+    private static final Map<String, VoucherType> VOUCHER_MAP =
+            Collections.unmodifiableMap(Stream.of(values()).collect(Collectors.toMap(VoucherType::getVoucherTypeString, Function.identity())));
+
+    VoucherType(String type) {
+        this.type = type;
+    }
+
+    public static VoucherType findVoucherMenu(String type) {
+        if (VOUCHER_MAP.containsKey(type)) {
+            return VOUCHER_MAP.get(type);
+        }
+
+        throw new IllegalArgumentException("Input: " + type + ", The type you are looking for is not found.");
+    }
+
+    public String getVoucherTypeString() {
+        return type;
+    }
+}
```

Comment:
다른 분 코드를 보다가 더 나은 방법이 있어서 추천드립니다.
현재 필드값에는 enum 타입을 소문자로 변경한 값이 들어가고 있는데, 이 경우 enum의 `valueOf` 메소드를 통해 심플하게 작성할 수 있더라구요~

```suggestion
public enum VoucherType {
    FIXED,
    PERCENT;

    VoucherType(String type) {
        this.type = type;
    }

    public static VoucherType findVoucherMenu(String type) {
        try {
            return valueOf(type.toUpperCase());
        } catch (Exception e) {
            new IllegalArgumentException("Input: " + type + ", The type you are looking for is not found.");
        }
    }

    public String getVoucherTypeString() {
        return type;
    }
}
```

Reply:
헉 안그래도!! 동일한 문자열이 들어가는거라 조금 비효율적이다라고도 생각이 들었는데 valueOf라는 메소드가 있군요!! 말씀해주신 코드에서 생성자랑 getter만 삭제하고 모든 enum클래스 다 동일하게 적용했습니다!! 👍 👍 감사합니다!!

적용 커밋: [1ffdd4a](https://github.com/prgrms-be-devcourse/springboot-basic/pull/675/commits/1ffdd4a3005daed7b1b7d071b6c6de6a91a09b96)